### ISSUE TITLE
Add operator support

### ISSUE THREAD

**Issue by brettlangdon** [brettlangdon/forge#14]
Idea revitalized from #9 

Originally I wanted to support operators like `+`, `-`, `/` and `*`, but cut it out of the original scope and now am considering adding them back in.

The simple use case here is for numbers.

```cfg
primary {
  num_workers: 2;
}
secondary {
  num_workers: 4;
}

total_workers: primary.num_workers + secondary.num_workers;
```

```cfg
one_hour: 3600;

primary {
  interval: 5 * one_hour;
}
```

As well, concatenation for strings would be helpful:

```cfg
prefix: "prefix:"

primary {
    string_setting: prefix + "primary";
}

secondary {
  string_setting: prefix + "secondary";
}
```

Where is starts to get trickier is with sections, do we merge them together?

```cfg
primary {
  some_setting: "value";
}

secondary {
  some_setting: "another value";
}

all_settings: primary + secondary;
```

What is the value of `all_settings` here?

Also, should consider this for lists when they are supported. Lists will probably be treated as sets and also support unions, intersections and etc

**Comment by lode** [comment]
I would say this belongs in the programming language using the configuration file, and not in the configuration file itself. A bit like the mustache mantra of "Logic-less templates". Keep it simple.
What's your stance in this?

**Comment by brettlangdon** [comment]
Yeah, I've gone back and forth on whether it is worth adding or not. Mostly wanted to put up these issues to see what people say.

**Comment by lode** [comment]
I do see the value in string concatenation (as in the referenced HOCON issue). Especially for i18n config that might be very handy. And number 'concatenation' seems not far away then, but calculating feels a bit off track for a config.

**Comment by brettlangdon** [comment]
Yeah, I am not 100% on the use case, just figured if I was going to do integer addition, wouldn't be too hard to add the others.
